core/src/transport/and_then.rs

```diff
@@ -78,7 +78,6 @@ where
                     role_override: Endpoint::Dialer,
                 },
             )),
-            _marker: PhantomPinned,
```

Comment:
Is this safe? I am sure this wasn't added without reason.

@mxinden Do you know why we phantom pin this type?